portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,57 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+document.addEventListener("DOMContentLoaded", function() { 
+  let currSlide = 1;
+  let numImgs = 3;
+
+  createSlide(currSlide, numImgs);
+
+  const prev = document.getElementById("prev");
+  const next = document.getElementById("next");
+
+  function changeSlide (direction) {
+    currSlide += direction;
+    if(currSlide < 1) {
+      currSlide = numImgs;
+    }
+    else if(currSlide > numImgs) {
+      currSlide = 1;
+    }
+    document.getElementById("slide-img").src = `images/img${currSlide}.jpg`;
+    document.getElementById("slide-number").textContent = `${currSlide} of ${numImgs}`;
+  }
+
+  prev.onclick = () => {
+    changeSlide(-1);
+  };
+
+  next.onclick = () => {
+    changeSlide(1);
+  };
+});
+
+function createSlide(currSlide, numImgs) {
```

Comment:
2 ideas:
(a) If you move this into the DOMContentLoaded scope, you'll get access to currSlide and numImgs, you can then make this function context-less.

(b) - preferred

consider doing the following:

```
document.addEventListener("DOMContentLoaded", function() { 
   const slides = new Slides(document.getElementById("slide"));
}

class Slides {
   constructor(slideElement) {
      this.curSlide = 1;
      this.numImgs = 3;
      this.slide = slideElement;
      this.prev = document.createElement('a');
      this.next = document.createElement('a');
      // everything else in this function, but use the "this" vars.
      
      // then register the listeners on prev/next here.
      this.prev.onclick = () => {
          this.changeSlide(-1);
      }
   }
   
   changeSlide (direction) { 
        // all the code you had before, but use the this vars.
   }
}

```